.github/workflows/ci.yml

```diff
@@ -44,7 +44,7 @@ on:
   pull_request:
 
 env:
-  NODE_VERSION: "14.x"
+  NODE_VERSION: "18.x"
```

Comment:
Is this required for this change? If we aren't upgrading the version, I'd prefer we keep it at the lowest supported so we don't miss issues.

Reply:
I had to upgrade ember-try after hitting https://github.com/ember-cli/ember-try/issues/967
> Run node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO --- ember test --environment=production
> Error!
> Error: Cannot copy '../../../browserslist/cli.js' to a subdirectory of itself, '../../../browserslist/cli.js'.
> Error: Cannot copy '../../../browserslist/cli.js' to a subdirectory of itself, '../../../browserslist/cli.js'.
>     at /home/runner/work/ember-data-model-fragments/ember-data-model-fragments/node_modules/ember-try/node_modules/fs-extra/lib/copy/copy.js:210:21
>     at FSReqCallback.oncomplete (fs.js:180:23)

And then ember-try v3 has a higher node requirement:
> error ember-try@3.0.0: The engine "node" is incompatible with this module. Expected version "16.* || >= 18.*". Got "14.21.3"

Node 16 was EOL in Sep 2023 so I went with v18.

If we want to stick with node 14, `yarn --ignore-engines` might work.

Reply:
I'm fine with upgrading to 18, we'd just need to update the readme and any other references, and do a major bump.

Reply:
`--ignore-engines` seems to work in CI. ember-try is a devDependency so it shouldn't affect the consuming app. I was able to install locally with node 14 and yarn 1.22 without needing to use `--ignore-engines`.